cmd/mkpkg/main.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// mkpkg builds the tailscale rpm and deb packages.
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/goreleaser/nfpm"
+	_ "github.com/goreleaser/nfpm/deb"
+	_ "github.com/goreleaser/nfpm/rpm"
+	"github.com/pborman/getopt"
+)
+
+func parseFiles(s string) (map[string]string, error) {
+	ret := map[string]string{}
+	for _, f := range strings.Split(s, ",") {
+		fs := strings.Split(f, ":")
+		if len(fs) != 2 {
+			return nil, fmt.Errorf("unparseable file field %q", f)
+		}
+		ret[fs[0]] = fs[1]
+	}
+	return ret, nil
+}
+
+func main() {
+	out := getopt.StringLong("out", 'o', "", "output file to write")
+	goarch := getopt.StringLong("arch", 'a', "amd64", "goarch this package is for")
+	pkgType := getopt.StringLong("type", 't', "deb", "type of package to build (deb or rpm)")
+	files := getopt.StringLong("files", 'F', "", "comma-separated list of files in src:dst form")
+	configFiles := getopt.StringLong("configs", 'C', "", "like --files, but for files marked as user-editable config files")
+	version := getopt.StringLong("version", 0, "0.0.0", "version of the package")
+	getopt.Parse()
+
+	filesMap, err := parseFiles(*files)
+	if err != nil {
+		log.Fatalf("Parsing --files: %v", err)
+	}
+	configsMap, err := parseFiles(*configFiles)
+	if err != nil {
+		log.Fatalf("Parsing --configs: %v", err)
+	}
+	info := nfpm.WithDefaults(&nfpm.Info{
+		Name:        "tailscale",
+		Arch:        *goarch,
+		Platform:    "linux",
+		Version:     *version,
+		Maintainer:  "Tailscale Inc <info@tailscale.com>",
+		Description: "The easiest, most secure, cross platform way to use WireGuard + oauth2 + 2FA/SSO",
```

Comment:
drop cross platform and oauth2 at least?

Reply:
Made the description a config param, in planning for a future where we build more than a "tailscale" package.